base/loading.jl

```diff
@@ -250,7 +250,7 @@ function locate_package(pkg::PkgId)::Union{Nothing,String}
                 # otherwise, signal failure
                 return implicit_manifest_uuid_path(env, pkg)
             end
-            @assert found.uuid !== nothing
+            found.uuid !== nothing || throw(ArgumentError("Package $pkg not found."))
```

Comment:
I believe this may be a correct assertion. @vtjnash, you refactored this code, can you take a look?

Reply:
It is correct (preserved from before the refactoring), although the behavior of this code-path is not documented.